assets/js/modules/adsense/components/dashboard/LegacyAdSenseDashboardWidgetOverview.js

```diff
@@ -85,9 +85,13 @@ export default function LegacyAdSenseDashboardWidgetOverview( props ) {
 			}
 		}
 	}, [
-		currentDataLoaded && previousDataLoaded, // All reports are fetched.
-		!! currentError || !! previousError, // Whether there is an error or not.
-		JSON.stringify( currentRangeData ),
+		currentDataLoaded,
+		currentError,
+		currentRangeData,
+		handleDataError,
+		handleDataSuccess,
+		previousDataLoaded,
+		previousError,
```

Comment:
As this is a `useEffect` this could be a cause for concern!

Note that `useEffect` re-runs when a dependency changes **not** when a dependency is truthy (the previous code does look a bit like it's expecting that)